lang/tests/src/test/scala/com/wavesplatform/lang/IntegrationTest.scala

```diff
@@ -1359,6 +1359,132 @@ class IntegrationTest extends PropSpec with PropertyChecks with ScriptGen with M
+          |let proof = base64'qV2FNaBFqWeL6n9q9OUbCSTcIQvwO0vfaA/f/SxEtLSIaOGIOx8r+WVGFdxmC6i3oOaoEkJWvML7PpKBDtqiK7pKDIaMV5PkV/kQl6UgxZv9OInTwpVPtYcgeeTokG/eBi1qKzJwDoEHVqKeLqrLXJHXhBVQLdoIUOeKj8YMkagVniO9EtK0fW0/9QnRIxXoilxSj5HBEpYwFBitJXRk1ftFGWZFxJXU5PXdRmC+pomyo5Scx+UJQ2NLRWHjKlV0'
+          |let inputs = base64'aZ8tqrOeEJKt4AMqiRF/WJhIKTDC0HeDTgiJVLZ8OEtiLNj7hflFeVnNXPguxyoqkI/V7pGJtXBpH5N+RswQNA0b23aM33aH0HKHOWoGY/T/L7TQzYFGJ3vTLiXDFZg1OVqkGOMvqAgonOrHGi6IgcALyUMyCKlL5BQY23SeILJpYKolybJNwJfbjxpg0Oz+D2fr7r9XL1GMvgblu52bVQT1fR8uCRJfSsgA2OGw6k/MpKDCfMcjbR8jnZa8ROEvF4cohm7iV1788Vp2/2bdcEZRQSoaGV8pOmA9EkqzJVRABjkDso40fnQcm2IzjBUOsX+uFExVan56/vl9VZVwB0wnee3Uxiredn0kOayiPB16yimxXCDet+M+0UKjmIlmXYpkrCDrH0dn53w+U3OHqMQxPDnUpYBxadM1eI8xWFFxzaLkvega0q0DmEquyY02yiTqo+7Q4qaJVTLgu6/8ekzPxGKRi845NL8gRgaTtM3kidDzIQpyODZD0yeEZDY1M+3sUKHcVkhoxTQBTMyKJPc+M5DeBL3uaWMrvxuL6q8+X0xeBt+9kguPUNtIYqUgPAaXvM2i041bWHTJ0dZLyDJVOyzGaXRaF4mNkAuh4Et6Zw5PuOpMM2mI1oFKEZj7Xqf/yAmy/Le3GfJnMg5vNgE7QxmVsjuKUP28iN8rdi4bUp7c0KJpqLXE6evfRrdZBDRYp+rmOLLDg55ggNuwog=='
+          |""".stripMargin)
+    )
+
+  property("groth16Verify_*inputs") {
+    for((ii, lets) <- grothsOk if (ii <= 16)) {
+      val i = if(ii == 0) { 1 } else { ii }
+      val src = lets ++ (if(i != 16) { s"groth16Verify_${i}inputs(vk, proof, inputs)" } else { "groth16Verify(vk, proof, inputs)" })
+      eval(src, version = V4) shouldBe Right(CONST_BOOLEAN(true))
+    }
+  }
+
+  property("groth16Verify_*inputs fail if too many inputs") {
+    for((i, lets) <- grothsFail ++ grothsOk if (i>1)) {
+      val src = lets ++ (if(i <= 16) { s"groth16Verify_${i-1}inputs(vk, proof, inputs)" } else { "groth16Verify(vk, proof, inputs)" })
+      if(i <= 16) {
+        eval(src, version = V4) shouldBe Left(s"Invalid inputs count $i, must be not greater than ${i-1}")
+      } else {
+        eval(src, version = V4) should produce("Groth16Verify key size should not exceed 1 Kbyte")
```

Comment:
1. Why such an error and not about count of inputs?
1. Why 1 Kbyte (1024 bytes) if `(8 + inputsCount)*48 = (8+16)*48 = 1152` bytes?
1. What if vk is fine, but inputs more than limit?